tests/test_skyBotEphemerisQuery.py

```diff
@@ -61,8 +62,7 @@ def requestReplace(input1, input2):
             """Junk wrapper for replacing the external internal call with an
             internel data load.
             """
-            with open(os.path.join(getPackageDir("ap_association"),
-                                   "tests",
+            with open(os.path.join(TESTDIR,
```

Comment:
Is there any particular reason to prefer the `__file__` approach other than convention?

Reply:
There are a couple of reasons why `__file__` is better.

* If the test data is in a subdir of the `tests` directory then you almost certainly want the data in that subdirectory to be the data that is used rather than whatever is pointed to by the setup package.
* `getPackageDir` relies on the environment variable being set (presumably by EUPS) and this is an unnecessary requirement when it comes to finding ones own test files in a subdirectory.

`getPackageDir` is used to find other packages and finding your own files relative to your own test directory shouldn't require EUPS. In an ideal world we would use `importlib.resources` to find data files and configs from other python packages but we haven't tried to tackle that yet.

I really thought we mentioned this in the dev guide but it seems there is no mention of how to find your own test files anywhere in the testing section.

Reply:
Sure, but you need the package to be set up anyway to be able to run the tests, so it seems to me like it amounts to the same thing. 🤷

Reply:
You could run the tests in a local directory but have the installed package be the one that is set up. Depends on what you are trying to do. Using `__file__` at least makes the tests be self-consistent without them causing confusion and finding the data from some other location.